taichi/ui/backends/vulkan/renderer.cpp

```diff
@@ -125,11 +126,14 @@ void Renderer::prepare_for_next_frame() {
 }
 
 void Renderer::draw_frame(Gui *gui) {
+  app_context_.prog()->synchronize();
```

Comment:
We need to check if the prog() is not nullptr. In case of AOT module, we don't have a program. I believe this is up to the host application to call synchronize() before the draw() call...

Reply:
nice catch